pkg/structs/flamebearer/flamebearer.go

```diff
@@ -101,24 +102,80 @@ type FlamebearerTimelineV1 struct {
 	Watermarks    map[int]int64 `json:"watermarks"`
 }
 
+type FlamebearerHeatmapV1 struct {
+	// Values matrix contain values that indicate count of value occurrences,
+	// satisfying boundaries of X and Y bins: [StartTime:EndTime) and (MinValue:MaxValue].
+	// A value can be accessed via Values[x][y], where:
+	//   0 <= x < TimeBuckets, and
+	//   0 <= y < ValueBuckets.
+	Values [][]uint64 `json:"values"`
+	// TimeBuckets denote number of bins on X axis.
+	// Length of Values array.
+	TimeBuckets int64 `json:"timeBuckets"`
+	// ValueBuckets denote number of bins on Y axis.
+	// Length of any item in the Values array.
+	ValueBuckets int64 `json:"valueBuckets"`
+	// StartTime and EndTime indicate boundaries of X axis: [StartTime:EndTime).
+	StartTime int64 `json:"startTime"`
+	EndTime   int64 `json:"endTime"`
+	// MinValue and MaxValue indicate boundaries of Y axis: (MinValue:MaxValue].
+	MinValue uint64 `json:"minValue"`
+	MaxValue uint64 `json:"maxValue"`
+	// MinDepth and MaxDepth indicate boundaries of Z axis: [MinDepth:MaxDepth].
+	// MinDepth is the minimal non-zero value that can be found in Values.
+	MinDepth uint64 `json:"minDepth"`
+	MaxDepth uint64 `json:"maxDepth"`
+}
```

Comment:
@dogfrogfog you can use this as a reference to the actual data models (hopefully, no more changes will be made in the spec). Also, I modified the way the grid is populated as per our discussion: now maximum Y value is at 0, and the minimum is the last element of a column